iron-swipeable-container.html

```diff
@@ -0,0 +1,266 @@
+        return;
+      this.unlisten(node, 'track', '_onTrack');
+      this.unlisten(node, 'transitionend', '_onTransitionEnd');
+    },
+
+    detached: function() {
+      if (this._nodeObserver) {
+        Polymer.dom(this.$.content).unobserveNodes(this._nodeObserver);
+      }
+    },
+
+    _onTrack: function(event) {
+      if (this.disableSwipe)
+        return;
+
+      var target = event.currentTarget;
+      if (target.classList.contains('disable-swipe'))
+        return;
+
+      var track = event.detail;
+      if (track.state === 'start') {
+        this._trackStart(track, target);
+      } else if (track.state === 'track') {
+        this._trackMove(track, target);
+      } else if (track.state === 'end') {
+        this._trackEnd(track, target);
+      }
+    },
+
+    _trackStart: function(event, target) {
+      // Save the width of the element, so that we don't trigger a style
+      // recalc every time we need it.
+      this._nodeWidth = target.offsetWidth;
+      target.style.transition = 'none';
+    },
+
+    _trackMove: function(event, target) {
+      this._animate(event.dx, target);
+    },
+
+    _trackEnd: function(event, target) {
+      // The element is swiped away if it's moved halfway its total width
+      this._swipeEnd(Math.abs(event.dx) > this._nodeWidth / 2, event.dx > 0, target);
+    },
+
+    _animate: function(x, target) {
+      var direction = x > 0 ? 1 : -1;
+
+      // This is the total distance the animation will take place over.
+      var totalDistance = this._nodeWidth * this.widthRatio;
+
+      // Opacity distance overflow. `this._nodeWidth * this.opacityRate` is the
+      // total distance the element needs to travel to become completely
+      // transparent, and `x` is how much the element has already travelled.
+      var opaqueDistance = Math.max(0, Math.abs(x) - this._nodeWidth * this.opacityRate);
+      var opacity = Math.max(0, (totalDistance - opaqueDistance) / totalDistance);
+      target.style.opacity = opacity
+
+      var translate, rotate;
+
+      if (this.swipeStyle === 'horizontal') {
+        translate = 'translate3d(' + x + 'px,' + 0 + 'px,0)';
+        rotate = '';
+      } else {  // Default is assumed to be `curve`.
+        // Assume the element will be completely transparent at 90 degrees, so
+        // figure out the rotation and vertical displacement needed to
+        // achieve that.
+        var y = totalDistance - Math.sqrt(totalDistance * totalDistance - opaqueDistance * opaqueDistance);
+        var deg = (1 - opacity) * direction * 90;
+
+        translate = 'translate3d(' + x + 'px,' + y + 'px,0)';
+        rotate = ' rotate(' + deg + 'deg)';
+      }
+
+      this.transform(translate + rotate, target);
+    },
+
+    _swipeEnd: function(away, dir, target) {
+      // Restore the original transition;
+      target.style.transition = this.transition;
+      this.away = away;
+      this.direction = dir ? 'right' : 'left';
```

Comment:
Are these deliberately public? If so, it might be good to pick names that give more context to their meaning.